pybossa/view/projects.py

```diff
@@ -1510,7 +1510,13 @@ def respond():
         args["records_per_page"] = per_page
         args["offset"] = offset
         start_time = time.time()
-        total_count, page_tasks = cached_projects.browse_tasks(project.get('id'), args, bool(args.get("filter_by_wfilter_upref")), current_user.id)
+        total_count, page_tasks = cached_projects.browse_tasks(
+            project.get('id'),
+            args,
+            bool(args.get("filter_by_wfilter_upref")),
+            current_user.id,
+            force_refresh=bool(args.get("filter_by_wfilter_upref"))
```

Comment:
1. can this variable be defined as `auto_refresh`. This also would require frontend changes
2. looks like same value passed at line 1516. Should this be instead checking for scheduler type
```python
auto_refresh = bool(project.get("info", {}).get("sched") == "task_queue")
total_count, page_tasks = cached_projects.browse_tasks(
     ...
     auto_refresh=auto_refresh)
```

Reply:
> 1. can this variable be defined as `auto_refresh`. This also would require frontend changes
> 2. looks like same value passed at line 1516. Should this be instead checking for scheduler type

1. This `force_refresh` is not the same as `auto-refresh` parameter on frontend and will not be passed to frontend. This parameter is to force `browse_tasks()` to recompute data, if `force_refresh` is not passed, the function would return cached data. 

2. Right, the same parameter is passed at line 1516, but line 1516 is passed as input to `browse_tasks()` while `force_refresh` is passed as input to memorization decorator. It is confusing to combine them together. 
Also, `force_refresh=true` only when project is on  task queue scheduler and the current user is a worker without any privilege.

Reply:
> This parameter is to force browse_tasks() to recompute data, if force_refresh is not passed, the function would return cached data.
I am also on same page as you on this and understand the usage of this parameter. As this would refresh page every 10 seconds, I interpret it as auto refresh every 10 seconds. I'm fine with `force_refresh` too. 

Regarding #2, isn't force/auto_refresh is to be set when scheduler is task queue, no? Wondering if we could just replace [this](https://github.com/bloomberg/pybossa/blob/1fb0120da0f76c76aae57f70e13f3d0600f0a900/pybossa/view/projects.py#L1487) with following

```python
args["force_refresh"] = True
```
and use following at line 1518
```python
force_refresh=bool(args.get("force_refresh"))
```

Reply:
We need this dict to generate sql filters for browse task page, we can not get rid of this code. https://github.com/bloomberg/pybossa/blob/1fb0120da0f76c76aae57f70e13f3d0600f0a900/pybossa/view/projects.py#L1487

everything in `args` are used for sql filter, so I prefer not to add `args["force_refresh"] = True`, or maybe we can add it first and pop out before entering `browse_tasks()`? Like 
```python
args["force_refresh"] = True
......
force_refresh = bool(args.pop("force_refresh"))
total_count, page_tasks = cached_projects.browse_tasks(
     ...
     auto_refresh= force_refresh)
```
what do you think?

Reply:
seems you don't need to put in args and pop it later. You can simply do
```python
force_refresh = bool(scheduler == Schedulers.task_queue)
total_count, page_tasks = cached_projects.browse_tasks(
     ...
     force_refresh=force_refresh)
```